wgsl/index.bs

```diff
@@ -344,10 +339,64 @@ Examples of predeclared objects are:
 * built-in types.
 
 A declaration must not introduce a name when that identifier is
-already in scope at the start of the declaration.
-That is, shadow names are not allowed in [SHORTNAME].
+already in scope with the same end scope as another instance of that name.
+When an identifier is referenced and multiple declarations are in scope, the
+reference is resolved to the closest preceding in scope declaration.
+
+<div class='example' heading='Valid and invalid declarations'>
+  <xmp>
+    // Invalid, cannot reuse built-in function names. See [[#builtin-functions]].
+    var<private> modf : f32 = 0.0;
+
+    // Valid, foo_1 is in scope until the end of the program.
+    var<private> foo : f32 = 0.0; // foo_1
+
+    // Valid, bar_1 is in scope until the end of the program.
+    var<private> bar : u32 = 0u; // bar_1
+
+    // Valid, my_func_1 is in scope until the end of the program.
+    // Valid, foo_2 is in scope until the end of the program.
```

Comment:
This is a typo.  foo2 is in scope until the end of the function.
(  Because it seems foo_3 correctly is not allowed to shadow.)